Formula/openssl110.rb

```diff
@@ -0,0 +1,184 @@
+    no-ssl3
+    no-ssl3-method
+    no-zlib
+    shared
+    enable-cms
+  ]
+  end
+
+  def install
+    if build.universal?
+      ENV.permit_arch_flags
+      archs = Hardware::CPU.universal_archs
+    elsif MacOS.prefer_64_bit?
+      archs = [Hardware::CPU.arch_64_bit]
+    else
+      archs = [Hardware::CPU.arch_32_bit]
+    end
+
+    dirs = []
+
+    archs.each do |arch|
+      if build.universal?
+        dir = "build-#{arch}"
+        dirs << dir
+        mkdir dir
+        mkdir "#{dir}/engines"
+      end
+
+      ENV.deparallelize
+      system "perl", "./Configure", *(configure_args + arch_args[arch])
+      system "make", "clean" if build.universal?
+      system "make", "depend"
+      system "make"
+      system "make", "test" if build.with?("test")
+
+      if build.universal?
+        cp "include/openssl/opensslconf.h", dir
+        cp Dir["*.?.?.dylib", "*.a", "apps/openssl"], dir
+        cp Dir["engines/**/*.dylib"], "#{dir}/engines"
+      end
+    end
+
+    system "make", "install", "MANDIR=#{man}", "MANSUFFIX=ssl"
+
+    if build.universal?
+      %w[libcrypto libssl].each do |libname|
+        system "lipo", "-create", "#{dirs.first}/#{libname}.1.1.dylib",
+                                  "#{dirs.last}/#{libname}.1.1.dylib",
+                       "-output", "#{lib}/#{libname}.1.1.dylib"
+        system "lipo", "-create", "#{dirs.first}/#{libname}.a",
+                                  "#{dirs.last}/#{libname}.a",
+                       "-output", "#{lib}/#{libname}.a"
+      end
+
+      Dir.glob("#{dirs.first}/engines/*.dylib") do |engine|
+        libname = File.basename(engine)
+        system "lipo", "-create", "#{dirs.first}/engines/#{libname}",
+                                  "#{dirs.last}/engines/#{libname}",
+                       "-output", "#{lib}/engines/#{libname}"
+      end
+
+      system "lipo", "-create", "#{dirs.first}/openssl",
+                                "#{dirs.last}/openssl",
+                     "-output", "#{bin}/openssl"
+
+      confs = archs.map do |arch|
+        <<-EOS.undent
+          #ifdef __#{arch}__
+          #{(buildpath/"build-#{arch}/opensslconf.h").read}
+          #endif
+          EOS
+      end
+      (include/"openssl/opensslconf.h").atomic_write confs.join("\n")
+    end
+  end
+
+  def openssldir
+    etc/"openssl110"
+  end
+
+  def post_install
+    # OpenSSL "remembers" the Perl it was built against. Where that is
+    # Homebrew's Perl it remembers the Cellar path, which breaks every version.
+    # Fix this manually to use the opt path which remains consistent.
+    perl_scripts = %W[
+      #{openssldir}/misc/CA.pl
+      #{openssldir}/misc/tsget
+      #{bin}/c_rehash
+    ]
+    perl_scripts.each do |f|
+      next unless File.exist?(f)
+      if File.read(f).include?(Formula["perl"].bin)
+        inreplace f, Formula["perl"].bin, Formula["perl"].opt_bin
+      end
+    end
+
+    keychains = %w[
+      /Library/Keychains/System.keychain
```

Comment:
@DomT4 Fair enough — but know this: I'm going to be offline for a bit.
Because, (I shit you not), someone already used this vector to plant some very gnarly malware 😞 on my machine. I literally just had this demo up for 8 hours; there was no server up before that.

----
From a tor exit node:
```
default 162.247.72.27 - - [08/May/2016:14:19:29 -0700] "GET / HTTP/1.1" 302 265 "http://burger-imperia.com/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/37.0.2062.120 Safari/537.36"
...
http://apple.com-systemnotice.com/eryp/index.html ... (includes my IP, geoip stuff, etc)... =BASE64dmlkLi4wMDAwMDAwNC0xYzIzLTQ5NDUtODAwMC0wMDAwMDAwMDAwMDBfX3ZwaWQuLjM1ODNmMDAwLTE1NjgtMTFlNi04NDdiLTc1MzY4ZDdmYTRmM19fY2FpZC4uMTQ5ZDNhYjMtNDg3Ny00N2M2LWEzYWEtYjJjYzE0YmFiYzA1X19ydC4uRF9fbGlkLi43NjUzYmZhNS0yNjA1LTQ1ZmMtYmQ0Yy1iNmJlMzA1MjgzZDZfX29pZDEuLmI4OGM4ZDlhLWNjMzktNDg2NC05ZDI0LWI0ZWQ5NWNmOThmN19fdmFyMS4udGFuZ28tcHllLVZmVThHdXlvX192YXIyLi50d2VlZGVoYW5kc19fdmFyNC4uTk9OLUFEVUxUX192YXI1Li5ET01BSU5fX3JkLi5fX2FpZC4uX19hYi4uX19zaWQuLg
...
vid..00000004-1c23-4945-8000-000000000000__vpid..3583f000-1568-11e6-847b-75368d7fa4f3__caid..149d3ab3-4877-47c6-a3aa-b2cc14babc05__rt..D__lid..7653bfa5-2605-45fc-bd4c-b6be305283d6__oid1..b88c8d9a-cc39-4864-9d24-b4ed95cf98f7__var1..tango-pye-VfU8Guyo__var2..tweedehands__var4..NON-ADULT__var5..DOMAIN__rd..__aid..__ab..__sid.
```

..and I'm still grepping through a billion logs for what happend after that, but somehow, I'm already on three blacklists already.

I had thought this would be a relatively benign way to demo this issue; but I guess not.
But please understand I intentionally omitted mentioning the real attack vectors in my commend above.

So, while you're investigating, you might wan't to use a VM.

I, (stupidly), just pointed my A record to my laptop.

So I'm gonna be doing some serious digging and reformatting; I may need sterilize the whole machine, depending on how deep this is. (I do have current backups, thankfully.)

So, a parting thought. Perhaps you may want to think of it this way: what relevant certificates exist in `/Library/Keychains/System.keychain` (any and all of which will be treated as root CAs) do we _need_ to have in `cert.pem`? What purpose do they serve? There is a reason the root CA list on operating systems are usually **locked down** (including OS X now, with SIU — but that doesn't cover `/Library/...`).

Anyway, if you need any pointers replicating/investigating, I'm happy to help, either here in this thread, and I can jump on IRC too if you give me some notice.

🍻